src/components/link/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { ElementType, FC, ReactElement, ReactNode } from 'react';
+import { chakra, useMultiStyleConfig } from '@chakra-ui/react';
+
+interface Props {
+  children: ReactNode;
+  as?: ElementType;
+  leftIcon?: ReactElement;
+  rightIcon?: ReactElement;
+  isExternal?: boolean;
+  [key: string]: unknown;
+}
+
+const Link: FC<Props> = ({
+  as = chakra.a,
+  children,
+  isExternal = false,
+  leftIcon,
+  rightIcon,
+  ...rest
+}) => {
+  const styles = useMultiStyleConfig(`Link`);
+
+  const Component = chakra(as, {
+    baseStyle: styles.link,
+  });
+
+  const textStyle = {
+    ...(leftIcon ? styles.leftIcon : {}),
+    ...(rightIcon ? styles.rightIcon : {}),
+  };
+
+  return (
+    <Component
+      target={isExternal ? '_blank' : undefined}
+      rel={isExternal ? 'noopener' : undefined}
```

Comment:
can we expose `rel` as a prop? There are cases where you want to add `nofollow` for internal links to not pass the SEO juice from valuable pages to non-valuable (and also external links)

Also, I would add `noreferrer` for external links as well because it has much better browser support than only `noopener` . What do you think?

Reply:
great 👍